### DISCUSSION TITLE
minimap2 - the road to long-read sequencing

### DISCUSSION THREAD

**Discussion by mdediegofuertes** [TORCH-Consortium/MAGMA#179] [Ideas]
Would it maybe make sense to add an option to use minimap2 instead of bwa-mem, in case we ever wanted to have a go at running long-read sequences through MAGMA?

https://lh3.github.io/2018/04/02/minimap2-and-the-future-of-bwa

**Comment by abhi18av** [comment]
Thanks for initiating this @mdediegofuertes , I'm also link a previous thread here https://github.com/TORCH-Consortium/MAGMA/issues/120 to bring to attention things we can expand upon.


Context for @TimHHH: 

- Since Prof. Annelies, Miguel and Vincent are here, I've bounced some ideas about the future roadmap for the pipeline. One of those ideas was to start allowing users to use this pipeline with long-read sequences.


- Would love to hear your thoughts on the topic of long read sequences.

**Reply by TimHHH** [reply]
Hey. Sorry for the late reply. Sure, I think it is a good idea to build in the functionality for nanopore sequences.
I think there are two ways of incorporating this idea:
1. try to get the nanopore data to a format compatible with the current pipeline as early as possible in the flow, e.g. produce a bam file for example with the suggested minimap and move it in to the normal workflow from thereon. Theoretically this should work with VQSR, but I don't think this is guaranteed or tested.
2. Do a completely seperate workflow for the long reads and gvcf merge into the regular workflow once reliable variants have been identified, i.e. after filtering.

The real nice thing about nanopore is off course that you could do de novo assemblies, with or without illumina, but this is not something that will work easily with MAGMA as a de novo genome is very difficult to align correctly with the reference genome.

Also, I can not immediately confirm if Minimap is appropriate for our illumina reads since we rely on the -k parameter for those. Minimap seems to have something similar (also -k), but this would require some parallel testing with BWA mem.

Tim

**Reply by abhi18av** [reply]
Thanks @TimHHH !

> ... try to get the nanopore data to a format compatible with the current pipeline as early as possible in the flow, e.g. produce a bam file for example with the suggested minimap and move it in to the normal workflow from thereon.

Actually this makes sense and I'd be happy to take this forward after some prelim. investigation has been done. 

To get started, perhaps either you or @mdediegofuertes  can get started with the experimentation outside of the pipeline using the sciensano samples and once we have a good direction regarding the filtering then we add that as an alternative workflow.

Or we could even add a flag `--experimental-long-reads` feature to make it explicit for any users.
